### DISCUSSION TITLE
How to configure "advanced" options for `airflow celery worker`?

### DISCUSSION THREAD

**Discussion by Gollum999** [apache/airflow#25918] [Q&A] [answered]
I am trying to set up some [configuration](https://docs.celeryq.dev/en/stable/userguide/configuration.html) options for my Celery workers, but `airflow.cfg` only exposes a limited number of options.  Some of the options I want to override are not exposed here.

I have tried `celery_config_options = MY_CUSTOM_CELERY_CONFIG`, and I see in the logs that this module is loaded when I do `airflow celery worker`, but none of my config overrides are applied to the worker process.  Looking through the code, it looks like this config is only used for `CeleryExecutor`, which as far as I can tell is not used at all by the worker processes.

How can I give `airflow celery worker` a custom config?

**Comment by potiuk** [comment, marked as answer]
Use CELERY_* environment variables.

https://stackoverflow.com/questions/53284139/can-i-use-environment-variables-to-make-celery-worker-connect-to-a-different-wor

**Comment by Gollum999** [comment]
Thanks @potiuk, I can work with this.  For posterity, I was able to use `celery_config_options = celery_config.CELERY_CONFIG` along with `export CELERY_CONFIG_MODULE=celery_config` to make the worker read and use the same config file.

One thing is a bit clunky: Airflow overrides a handful of config options by passing them in as command line args.  Environment variable overrides don't work for these options because the command line *always* takes precedence.  As a result, options like `task_queues` or `worker_concurrency` *must* be specified on the command line or in `airflow.cfg` instead of in the celery config file.

**Reply by potiuk** [reply]
PR's to fix that behaviour are most welcome :)

**Reply by vaishali-cm** [reply]
Was this resolved?
